crates/matrix-sdk-sled/src/encode_key.rs

```diff
@@ -39,6 +39,12 @@ impl<T: EncodeKey + ?Sized> EncodeKey for &T {
     }
 }
 
+impl EncodeKey for &[u8] {
+    fn encode_as_bytes(&self) -> Cow<'_, [u8]> {
+        (*self).into()
```

Comment:
Do we really want this impl? The slice could contain our `ENCODE_SEPARATOR`.

Reply:
hmm.... while that is a fair argument in general, this is only used (and usable for the outside) via the `custom`-feature-are and there we are a) in our own bucket and b) don't (yet) use the encode separator. 

However, I agree that this is a very generic implementation and not safe as it is. I will give it a wrapper type that one has to use instead.